docs/integrations/channels.md

```diff
@@ -0,0 +1,209 @@
+---
+title: Channels
+---
+
+# Channels
+
+Strawberry provides support for [Channels](https://channels.readthedocs.io/)
+with
+[Consumers](https://channels.readthedocs.io/en/stable/topics/consumers.html) to
+provide GraphQL support over WebSockets and HTTP.
+
+While Channels does require Django to be installed as a dependency, you can
+actually run this integration without using Django's request handler. However
+the most common use case will be to run a normal Django project with GraphQL
+subscriptions support, typically taking advantage of the Channel Layers
+functionality which is exposed through the Strawberry integration.
+
+## Getting Started
+
+Before using Strawberry's Channels support, make sure you install all the
+required dependencies by running:
+
+```
+pip install 'strawberry-graphql[channels]'
+```
+
+The example below assumes you've already got a working Django project, having
+followed the
+[installation instructions](https://channels.readthedocs.io/en/stable/installation.html)
+for Channels, and while not absolutely required, you probably want to also have
+set up Django integration for Strawberry by following the
+[documentation here](../integrations/django.md).
+
+The easiest way to use Channels with Strawberry is to use the
+GraphQLProtocolTypeRouter, which will wrap your Django application, and route
+HTTP and websockets for /graphql to Strawberry, while sending all other requests
+to Django. You'll need to modify the `myproject.asgi.py` file from the Channels
+instructions to look something like this:
+
+```python
+import os
+
+from django.core.asgi import get_asgi_application
+from strawberry.channels import GraphQLProtocolTypeRouter
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'mysite.settings')
+django_asgi_app = get_asgi_application()
+
+# Import your Strawberry schema after creating the django ASGI application
+# This ensures django.setup() has been called before any ORM models are imported
+# for the schema.
+from mysite.graphql import schema
+
+
+application = GraphQLProtocolTypeRouter(
+    schema,
+    django_application=django_asgi_app,
+)
+```
+
+The above code is not very flexible, taking away some of the useful capabilities
+of Channels. For more complex deployments, you'll most likely want to read on
+further...
+
+## Custom ProtocolTypeRouter
+
+A more complex use case is when you want to integrate several ASGI applications
+on different URLs and protocols, like what is described in the
+[Channels documentation](https://channels.readthedocs.io/en/stable/topics/protocols.html).
+
+An example of this would be:
+
```

Comment:
```suggestion
```py
from channels.auth import AuthMiddlewareStack
from channels.routing import ProtocolTypeRouter, URLRouter
from channels.security.websocket import AllowedHostsOriginValidator
from django.core.asgi import get_asgi_application
from django.urls import re_path
from strawberry.channels import GraphQLHTTPConsumer, GraphQLWSConsumer

django_asgi_app = get_asgi_application()
from tserver.chat import routing
from tserver.tzv5hob.schemas import schema

# Import your Strawberry schema after creating the django ASGI application
# This ensures django.setup() has been called before any ORM models are imported
# for the schema.

application = ProtocolTypeRouter(
    {
        "http": URLRouter(
            [
                re_path(  # Here you will interact with your schema
                    "^graphql", AuthMiddlewareStack(GraphQLHTTPConsumer.as_asgi(schema=schema))
                ),
                re_path("^", django_asgi_app),  # This might be another endpoints in your app
            ]
        ),
        "websocket": URLRouter(
            [
                re_path("^graphql", AuthMiddlewareStack(GraphQLWSConsumer.as_asgi(schema=schema))),
                re_path(
                    "chat",  
                    # This can be another WS endpoint, i,e the chat example from channels docs.
                    AllowedHostsOriginValidator(
                        AuthMiddlewareStack(URLRouter(routing.websocket_urlpatterns))
                    ),
                ),
            ]
        ),
    }
)

```
```